internal/controller/kyma/controller.go

```diff
@@ -209,6 +207,42 @@ func (r *Reconciler) reconcile(ctx context.Context, kyma *v1beta2.Kyma) (ctrl.Re
 	return res, err
 }
 
+func (r *Reconciler) replaceKymaFromRemote(ctx context.Context, kyma *v1beta2.Kyma) ReconciliationLoopResult {
+	updateRequired, err := r.SyncRemoteCrds.Execute(ctx, kyma)
+	if err != nil {
+		r.Metrics.RecordRequeueReason(metrics.CrdsSync, queue.UnexpectedRequeue)
+		return newReconcileResult(r.requeueWithError(ctx, kyma, fmt.Errorf("could not sync CRDs: %w", err)))
+	}
+	if updateRequired {
+		if err := r.Update(ctx, kyma); err != nil {
+			r.Metrics.RecordRequeueReason(metrics.CrdAnnotationsUpdate, queue.UnexpectedRequeue)
+			return newReconcileResult(r.requeueWithError(ctx, kyma, fmt.Errorf("could not update kyma annotations: %w", err)))
+		}
+		r.Metrics.RecordRequeueReason(metrics.CrdAnnotationsUpdate, queue.IntendedRequeue)
+		return newReconcileResult(ctrl.Result{Requeue: true}, nil)
+	}
+	// update the control-plane kyma with the changes to the spec of the remote Kyma
+	if err = r.replaceSpecFromRemote(ctx, kyma); err != nil {
+		if errors.Is(err, ErrInvalidKymaSpec) {
+			// if the spec is invalid, we should not continue with processing, but make the user aware of the issue
+			if sErr := r.updateStatusWithError(ctx, kyma, err); sErr != nil {
```

Comment:
please don't use `updateStatusWithError` here, we have the pattern always update status when requeue happens, that's why before your implementation, the only usage of `updateStatusWithError` is in `requeueWithError`.

Reply:
update: I don't think you need to change this existing logic https://github.com/kyma-project/lifecycle-manager/pull/1748#discussion_r1711371963

Reply:
I'll try to to that.

Reply:
reconcile logic reverted.